File: frontend/src/types/consultation.ts
export interface Consultation {
  id: number;
  consultationId: string;
  patientId: number;
  patientName: string;
  doctorId: number;
  doctorName: string;
  branchId: number;
  branchName: string;
  consultationDate: string;
  symptoms: string;
  clinicalNotes: string;
  diagnosis?: string;
  prescription?: string;
  labRequestStatus?: string;
  imagingRequestStatus?: string;
  followUpRequired?: boolean;
  followUpDate?: string;
}

export interface CreateConsultationRequest {
  patientId: number;
  doctorId: number;
  branchId: number;
  symptoms: string;
  clinicalNotes?: string;
  diagnosis?: string;
  prescription?: string;
  labRequestStatus?: string;
  imagingRequestStatus?: string;
  followUpRequired?: boolean;
  followUpDate?: string;
}

File: backend/src/main/java/com/hospital/consultation/model/Consultation.java
package com.hospital.consultation.model;

import com.hospital.auth.model.User;
import com.hospital.multibranch.model.Branch;
import com.hospital.patient.model.Patient;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "consultations")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Consultation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "consultation_id", unique = true, length = 50)
    private String consultationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", nullable = false)
    private Patient patient;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", nullable = false)
    private User doctor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(name = "consultation_date")
    private LocalDateTime consultationDate;

    @Column(columnDefinition = "TEXT")
    private String symptoms;

    @Column(columnDefinition = "TEXT")
    private String clinicalNotes;

    @Column(columnDefinition = "TEXT")
    private String diagnosis;

    @Column(columnDefinition = "TEXT")
    private String prescription;

    @Column(name = "lab_request_status")
    private String labRequestStatus;

    @Column(name = "imaging_request_status")
    private String imagingRequestStatus;

    @Column(name = "follow_up_required")
    private Boolean followUpRequired = false;

    @Column(name = "follow_up_date")
    private LocalDateTime followUpDate;

    @Column(name = "is_active")
    @Builder.Default
    private Boolean isActive = true;

    @Column(name = "created_at", updatable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    @Builder.Default
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        this.consultationDate = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
